### DISCUSSION TITLE
Changing the Window theme dynamically

### DISCUSSION THREAD

**Discussion by xirano77** [appcelerator/titanium_mobile#13744] [Q&A] [answered]
Hello. I created two files for my dark and light theme `MyDarkTheme.xml` and `MyLightTheme.xml`. 
By default, I have a light theme, but I want it to change to dark when I click on the button.

My XML code (index.xml):
```xml
<Alloy>
  <Window id="win" theme='Theme.MyLightTheme'>
    <Button onClick="setDark">Set Dark</Button>
  </Window>
</Alloy>
```

My controller (index.js):
```js
$.win.open()

function setDark() {
    $.win.applyProperties({theme: 'Theme.MyDarkTheme'})
}
``` 

But dynamically it is not possible to change the theme. How can i do this? I tried searching for a similar question but couldn't find it.

**Comment by m1ga** [comment, marked as answer]
You have to reopen the window. Check the docs: https://titaniumsdk.com/api/titanium/ui/window.html#properties_theme it says "creation only". Those properties only work one time when the window is created.

That said: you can switch between the system dark/light mode. Check the `extended app sample` in [this comment](https://github.com/tidev/titanium_mobile/pull/13265#issuecomment-1035132153) (you have to toggle it). That way you would need to create a standard Android dark theme (same theme name in a `values-dark` folder)

**Reply by xirano77** [reply]
Thank you
